static/js/auth.ts

```diff
@@ -362,7 +362,7 @@ $ ('#email, #mail_repeat').on ('cut copy paste', function (e) {
  * - Check if we were supposed to be joining a class. If so, join it.
  * - Otherwise redirect to "my programs".
  */
-async function afterLogin() {
+async function afterLogin(teacher: boolean) {
```

Comment:
Nitpicky; I think we should pass an object to `afterLogin` - if we ever want to have different behaviours after logging in it may become a bit confusing in which order we pass the "login state";
 
 I propose:

```typescript
afterLogin({"teacher": response['teacher']})
```